test/CodeGenILTests/ReplicationTests.cs

```diff
@@ -678,5 +678,45 @@ public void MSIL_FuncCall_Double_SomeGuides()
                 },
                 output["test4"]));
         }
+
+        [Test]
+        public void Value_Are_Correct()
+        {
+            string code =
+            @"
+            import(""DesignScriptBuiltin.dll"");
+            import(""DSCoreNodes.dll"");
+
+            a = 0.5;
+            b = DSCore.Math.Sum([0.0, 10.0]);
+            ";
+
+            var ast = ParserUtils.Parse(code).Body;
+            var output = codeGen.EmitAndExecute(ast);
+            Assert.IsNotEmpty(output);
+            Assert.AreEqual(0.5, output["a"]);
+            Assert.AreEqual(10.0, output["b"]);
+
+        }
+        [Test]
+        [Category("Failure")] // Fails because we use the wrong type for a
```

Comment:
Can you explain what you mean by "we use the wrong type for a"

Reply:
We use 'object' instead of `double` because of the bug.

Reply:
I will update the comment.